feature/experimental/p4rt/ate_tests/lldp_packetin_test/lldp_packetin_test.go

```diff
@@ -0,0 +1,510 @@
+}
+
+// configureATE configures port1 and port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	i1 := top.AddInterface(atePort1.Name).WithPort(p1)
+	i1.IPv4().
+		WithAddress(atePort1.IPv4CIDR()).
+		WithDefaultGateway(dutPort1.IPv4)
+
+	p2 := ate.Port(t, "port2")
+	i2 := top.AddInterface(atePort2.Name).WithPort(p2)
+	i2.IPv4().
+		WithAddress(atePort2.IPv4CIDR()).
+		WithDefaultGateway(dutPort2.IPv4)
+
+	return top
+}
+
+// configureDeviceID configures p4rt device-id on the DUT.
+func configureDeviceID(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
+	nodes := p4rtutils.P4RTNodesByPort(t, dut)
+	p4rtNode, ok := nodes["port1"]
+	if !ok {
+		t.Fatal("Couldn't find P4RT Node for port: port1")
+	}
+	t.Logf("Configuring P4RT Node: %s", p4rtNode)
+
+	c := oc.Component{}
+	c.Name = ygot.String(p4rtNode)
+	c.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	c.IntegratedCircuit.NodeId = ygot.Uint64(deviceID)
+	gnmi.Replace(t, dut, gnmi.OC().Component(p4rtNode).Config(), &c)
+}
+
+// configurePortID configures p4rt port-id on the DUT.
+func configurePortID(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
+	ports := sortPorts(dut.Ports())
+	for i, port := range ports {
+		gnmi.Replace(t, dut, gnmi.OC().Interface(port.Name()).Id().Config(), uint32(i)+portID)
+	}
+}
+
+// setupP4RTClient sends client arbitration message for both leader and follower clients,
+// then sends setforwordingpipelineconfig with leader client.
+func setupP4RTClient(ctx context.Context, args *testArgs) error {
+	// Setup p4rt-client stream parameters
+	streamParameter := p4rt_client.P4RTStreamParameters{
+		Name:        streamName,
+		DeviceId:    deviceID,
+		ElectionIdH: uint64(0),
+		ElectionIdL: electionID,
+	}
+
+	// Send ClientArbitration message on both p4rt leader and follower clients.
+	clients := []*p4rt_client.P4RTClient{args.leader, args.follower}
+	for index, client := range clients {
+		if client != nil {
+			client.StreamChannelCreate(&streamParameter)
+			if err := client.StreamChannelSendMsg(&streamName, &p4v1.StreamMessageRequest{
+				Update: &p4v1.StreamMessageRequest_Arbitration{
+					Arbitration: &p4v1.MasterArbitrationUpdate{
+						DeviceId: streamParameter.DeviceId,
+						ElectionId: &p4v1.Uint128{
+							High: streamParameter.ElectionIdH,
+							Low:  streamParameter.ElectionIdL - uint64(index),
+						},
+					},
+				},
+			}); err != nil {
+				return errors.New("errors seen when sending ClientArbitration message")
+			}
+			if _, _, arbErr := client.StreamChannelGetArbitrationResp(&streamName, 1); arbErr != nil {
+				if err := p4rtutils.StreamTermErr(client.StreamTermErr); err != nil {
+					return err
+				}
+				return errors.New("errors seen in ClientArbitration response")
+			}
+		}
+	}
+
+	// Load p4info file.
+	p4Info, err := utils.P4InfoLoad(p4InfoFile)
```

Comment:
assignment copies sync.Mutex by value

Reply:
Discussed this offline with Moji/Vibov, this change will be addresses in another PR, since it involves more detailed changes to the function.